Automattic/woocommerce-payments
# Show active dispute as inline notice in order edit screen

Fixes: #6354

See also: https://github.com/woocommerce/woocommerce/pull/38443

#### Changes proposed in this Pull Request

This PR uses a new hook `woocommerce_admin_order_data_after_payment_info` to add a dispute notice to order edit page, inside the order details box. The tracks event `wcpay_order_dispute_notice_action_click` will be called when clicking the notice action button.

* ~It also uses the WCPay `InlineNotice` component which is a stand-in / prototype for a component proposed for WC core.~
	* **Update:** the component we're looking to utilise for this design has already been implemented as `BannerNotice` in #5814.
* Colour and wording **Note these differ from the design below:**
	* \> 7 days – 🟡 Yellow, `This order has a chargeback dispute of {AMOUNT} labeled as {REASON}. Please respond to this dispute before {DEADLINE}`
	* < 7 days – 🟡 Yellow, `Please resolve the dispute on this order for {AMOUNT} labeled {REASON} by {DEADLINE}. ({N} days left)`
	* < 72hours – 🔴 Red, `Please resolve the dispute on this order for {AMOUNT} labeled {REASON} by {DEADLINE}. ({N} days left)`
	* < 24hours – 🔴 Red, `Please resolve the dispute on this order for {AMOUNT} labeled {REASON} by {DEADLINE}. (Last day today)`

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/0e087701-6681-4c1b-9d85-c02e0bccbf1c)

#### Screenshots of this PR's implementation
\> 7 days
<img width="963" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/ecd51feb-d085-4aa1-bb55-179673ec8141">

< 7 days
<img width="968" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/e72551c1-b92a-4f76-bc8f-9f8e0ec7b7c4">

< 72 hours
<img width="965" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/8b684a9e-7394-46d4-83e8-01875d5056d9">

< 24 hours
<img width="977" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/7a4c95a3-6f8c-41bb-b108-69fe83c2061b">

 
## Next steps
- [x] Preferably, move changes to `InlineNotice` to a new PR to simplify testing and review.
See #6535.
- [x] Confirm `BannerNotice` is the correct component to utilise for this design.
- [x] Render date and do date logic 100% on client. I'm rendering on the server, but the date should be passed down in binary / unformatted, and then the client can do all the rendering and urgency calculations. This is not absolutely essential, but seems best to me. Note that `moment` is throwing an error due to the current sloppy approach (`value provided is not in a recognized RFC2822 or ISO format`).
   - **Note** from @jinksi: we'll be sharing more logic across FE components (e.g. `isDisputeUrgent()` to determine yellow or red colour) in Phase 2, so lean towards sharing in FE where suitable. See paJDYF-9oX-p2
- [x] Only show notice when there is an active dispute. I haven't tested this, will need to ensure notice is not shown when merchant has responded (etc).
- [x] Test with WC < 7.9 (before hook `woocommerce_admin_order_data_after_payment_info` added)

###### Follow ups
These tasks don't block this PR but would be good to follow up (e.g. log issues).

- [ ] When working on this I discovered that we have a few hard-coded `/ 100` (stripe value => currency) in the code base. These should use a convert utility that knows about zero decimal currencies. Maybe also `* 100`, see also specific issue #4345.

#### Testing instructions

- Install and activate WC 7.9+
- Set up a dispute by purchasing with a [dispute test card](https://stripe.com/docs/testing#disputes).
- View the order edit page and confirm the alert shows correct info and button links to dispute detail. 
- Try various combinations of dispute amount, date, and store settings (e.g. multicurrency) to ensure the notice appears when necessary and renders correctly.
- Ensure the notice is not displayed if the dispute has been lost or otherwise resolved, or the merchant has already submitted evidence (responded).

-------------------


**Post merge**
